tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala

```diff
@@ -173,6 +177,7 @@ class Z3SolverContext(val config: SolverConfig) extends SolverContext with LazyL
     state = Disposed()
     // Try to obtain the lock, to let the statistics thread finish its work.
     // If it is stuck for some reason, continue after the timeout in any case.
+    statisticsThread.interrupt()
     statisticsLock.tryLock(2 * config.z3StatsSec, java.util.concurrent.TimeUnit.SECONDS)
```

Comment:
Since you're waiting `2*z3StatsSec` for the lock, why do you need the interrupt?

I can see how it shortcuts while the thread is sleeping. But then you shouldn't need to wait `2*z3StatsSec` afterwards.

Also, what happens if you interrupt it while it's not sleeping?
From the [javadoc](https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#interrupt--) it does not look like it'd have any effect.